test/extended/router/externalCertificate.go

```diff
@@ -0,0 +1,682 @@
+package router
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	routev1 "github.com/openshift/api/route/v1"
+	"github.com/openshift/origin/test/extended/router/certgen"
+	exutil "github.com/openshift/origin/test/extended/util"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apimachinery/pkg/util/wait"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/framework/pod"
+)
+
+const (
+	// SECRET_READER_ROLE is the name of the Role allowing access to the secret.
+	SECRET_READER_ROLE = "secret-reader-role"
+	// SECRET_READER_ROLE_BINDING is the name of the RoleBinding associating the Role with the router service account.
+	SECRET_READER_ROLE_BINDING = "secret-reader-role-binding"
+)
+
+var _ = g.Describe("[sig-network][OCPFeatureGate:RouteExternalCertificate][Feature:Router][apigroup:route.openshift.io]", func() {
+	defer g.GinkgoRecover()
+	var (
+		oc            = exutil.NewCLIWithPodSecurityLevel("router-external-certificate", admissionapi.LevelBaseline)
+		helloPodPath  = exutil.FixturePath("..", "..", "examples", "hello-openshift", "hello-pod.json")
+		helloPodName  = "hello-openshift"
+		helloPodSvc   = "hello-openshift"
+		defaultDomain string
+		err           error
+	)
+
+	g.BeforeEach(func() {
+		defaultDomain, err = getDefaultIngressClusterDomainName(oc, time.Minute)
+		o.Expect(err).NotTo(o.HaveOccurred(), "failed to find default domain name")
+
+		g.By("creating pod")
+		err = oc.Run("create").Args("-f", helloPodPath, "-n", oc.Namespace()).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("waiting for the pod to be running")
+		err = pod.WaitForPodNameRunningInNamespace(context.TODO(), oc.KubeClient(), helloPodName, oc.Namespace())
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("creating service")
+		err = oc.Run("expose").Args("pod", helloPodName, "-n", oc.Namespace()).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("waiting for the service to become available")
+		err = exutil.WaitForEndpoint(oc.KubeClient(), oc.Namespace(), helloPodSvc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+
+	g.Context("with invalid setup", func() {
+		var host string
+		g.BeforeEach(func() {
+			host = oc.Namespace() + "." + defaultDomain
+		})
+
+		g.Describe("the router", func() {
+			g.It("should not support external certificate without proper permissions", func() {
+				g.By("Creating a secret with crt/key")
+				secret, _, err := createSecret(oc.Namespace(), "my-tls-secret", corev1.SecretTypeTLS, host)
```

Comment:
where the cleanup for `${oc.Namespace()}/my-tls-secret` Secret?
I see the same object is being created a few times.

Reply:
briefly recall some tooling/wrappers exist which auto-creates a new namespace when a portion of the test starts, so each time there is a new namespace for `oc.Namespace()`.
is my understanding correct?

And, that'd suggest that the `createNamespace` is only for creating the extra `differentNamespace`.

Reply:
Right, there is some wrapper which does the cleanup for all the resources created and it will delete the `oc.Namespace()` as well. Each test executes in a self-contained manner is what I've understood.